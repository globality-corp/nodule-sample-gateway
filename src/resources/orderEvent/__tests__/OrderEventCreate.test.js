import request from 'supertest';
import { Nodule } from '@globality/nodule-config';
import { signSymmetric } from '@globality/nodule-graphql';
import { mockResponse } from '@globality/nodule-openapi';
import createApp from '../../../app';

let app;

const email = 'user@globality.com';
const token = signSymmetric({ email });

beforeEach(async () => {
    await Nodule.testing().fromObject(
        mockResponse('charmander', 'orderEvent.create', {
            id: '00000000-0000-0000-0000-000000000000',
            customerId: '00000000-0000-0000-0000-111111111111',
            eventType: 'PizzaCreated',
            orderId: '00000000-0000-0000-0000-222222222222',
            pizzaSize: 'REGULAR',
            crustType: 'REGULAR',
            toppingType: null,
        }),
    ).load();

    app = createApp();
});

it('should create an order event', async () => {
    const query = `
        mutation createOrderEvent($input: OrderEventCreateInputType) {
          orderEventCreate(input: $input) {
            id
            eventType
          }
        }
    `;

    const variables = {
        input: {
            customerId: '00000000-0000-0000-0000-111111111111',
            eventType: 'PizzaCreated',
            orderId: '00000000-0000-0000-0000-222222222222',
            pizzaSize: 'REGULAR',
            crustType: 'REGULAR',
        },
    };


    const result = await request(app).post(
        '/gql/graphql',
    ).set(
        'Authorization', `Bearer ${token}`,
    ).send({
        query,
        variables,
    });

    expect(result.body.errors).toBeUndefined();
    expect(result.body).toMatchSnapshot();
});
