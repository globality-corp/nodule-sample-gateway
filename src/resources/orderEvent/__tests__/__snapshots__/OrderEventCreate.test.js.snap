// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create an order event 1`] = `
Object {
  "data": Object {
    "orderEventCreate": Object {
      "eventType": "PizzaCreated",
      "id": "00000000-0000-0000-0000-000000000000",
    },
  },
}
`;
